src/state/selectors/selectors.ts

```diff
@@ -209,7 +210,12 @@ export const combineTooltipPayload = (
     // BaseAxisProps does not support nameKey but it could!
     const finalNameKey: DataKey<any> | undefined = settings?.nameKey; // ?? tooltipAxis?.nameKey;
     let tooltipPayload: unknown;
-    if (tooltipAxis?.dataKey && !tooltipAxis?.allowDuplicatedCategory && Array.isArray(sliced)) {
+    if (
```

Comment:
Too much coincidence in one `if` for my liking. But the storybook works well and all tests are passing so